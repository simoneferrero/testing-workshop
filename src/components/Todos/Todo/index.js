import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

import { SPACING } from 'constants/index'

const StyledTodo = styled.button`
	background-color: inherit;
	border: none;
	color: ${({ completed }) => completed && '#808080'};
	cursor: pointer;
	font-family: 'Nanum Pen Script', cursive;
	font-size: 24px;
	margin-bottom: ${SPACING.half}px;
	padding: ${SPACING.half}px ${SPACING.base}px;
	text-align: left;
	text-decoration: ${({ completed }) => completed && 'line-through'};
	transition: box-shadow 0.3s ease-in-out, color 0.3s ease-in-out,
		transform 0.3s ease-in-out;
	width: 100%;
	will-change: transform;

	:hover {
		box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.3);
		transform: translate(5px, -5px) scale(1.01);
	}
`

const Todo = ({ completed, handleClick, id, text }) => (
	<StyledTodo completed={completed} onClick={() => handleClick(id, !completed)}>
		{text}
	</StyledTodo>
)

Todo.propTypes = {
	completed: PropTypes.bool.isRequired,
	handleClick: PropTypes.func.isRequired,
	id: PropTypes.string.isRequired,
	text: PropTypes.string.isRequired,
}

export default Todo
