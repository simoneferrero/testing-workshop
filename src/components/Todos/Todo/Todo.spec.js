import Todo from './index'

describe('Given <Todo />', () => {
	it('should render correctly', () => {
		const props = {
			completed: false,
			handleClick: jest.fn(),
			id: '12345',
			text: 'Some Todo',
		}
		const { container } = render(<Todo {...props} />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
