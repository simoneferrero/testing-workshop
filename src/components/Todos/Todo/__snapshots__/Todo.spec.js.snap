// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Todo /> should render correctly 1`] = `
<button
  class="Todo__StyledTodo-sc-1u0d2ms-0 gdiHPr"
  data-testid="todo-12345"
>
  Some Todo
</button>
`;
